pytorch_lightning/utilities/upgrade_checkpoint.py

```diff
@@ -0,0 +1,45 @@
+import argparse
+from shutil import copyfile
+
+import torch
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
+
+KEYS_MAPPING = {
+    "checkpoint_callback_best_model_score": (ModelCheckpoint, "best_model_score"),
+    "checkpoint_callback_best_model_path": (ModelCheckpoint, "best_model_path"),
+    "checkpoint_callback_best": (ModelCheckpoint, "best_model_score"),
+    "early_stop_callback_wait": (EarlyStopping, "wait_count"),
+    "early_stop_callback_patience": (EarlyStopping, "patience"),
+}
+
+
+def upgrade_checkpoint(filepath):
+    checkpoint = torch.load(filepath)
+    checkpoint["callbacks"] = checkpoint.get("callbacks") or {}
+
+    for key, new_path in KEYS_MAPPING.items():
+        if key in checkpoint:
+            value = checkpoint[key]
+            callback_type, callback_key = new_path
+            checkpoint["callbacks"][callback_type] = checkpoint["callbacks"].get(callback_type) or {}
+            checkpoint["callbacks"][callback_type][callback_key] = value
+            del checkpoint[key]
+
+    torch.save(checkpoint, filepath)
+
+
+if __name__ == "__main__":
```

Comment:
I am not very fun of having main in the package, especially if it is not the main entry point...
I see the reason for it, also understand to keep it in package, but...
anyways is it also tested as to be called as a script?

Reply:
I'm certainly open to better alternatives, do you have any ideas? I feel like when we return the error to upgrade the checkpoint our users will want to be able to quickly run the line we give them and then re-run their original train script.

I pulled most of the logic out in functions which are tested but the actual script is not.